### PR TITLE
Improve json read and write

### PR DESCRIPTION
**Pull request by AtuyL**
- changed to parallel writing.
   this is similar by https://github.com/dat-land/dat-desktop/commit/2a28c090269b01c77c4f19e6512083b2cc994f2c
- fix block the twice parse for json.

and minimal linting.

### CODE REVIEW

File: app/actions/dat-middleware.js [hunk 1]
```diff
@@ -387,19 +387,29 @@ export default class DatMiddleware {
...
-      const opts = JSON.parse(datOpts[key])
-      this.internalAddDat({
-        key,
-        path: opts.dir,
-        paused: paused[key],
-        ...opts
-      })
+      let datdata
+
+      // It is for compatibility.
```
Comment (martinheidegger) [inline app/actions/dat-middleware.js]:
```
// Backwards compatibility:
// In releases previous to 3.0.1 beta, the connection-data for each 
// DAT was additionally stringified. With this check we make sure
```

File: app/actions/dat-middleware.js [hunk 2]
```diff
@@ -387,19 +387,29 @@ export default class DatMiddleware {
...
+      try {
+        datdata = JSON.parse(datOpts[key])
```
Comment (martinheidegger) [inline app/actions/dat-middleware.js]: Why did you decide to use a try-catch here? It is very slow; a `typeof` test would have been faster and it wouldn't catch other weird cases.

File: app/actions/dat-middleware.js [hunk 3]
```diff
@@ -387,19 +387,29 @@ export default class DatMiddleware {
...
 
     for (const key of Object.keys(datOpts)) {
-      const opts = JSON.parse(datOpts[key])
-      this.internalAddDat({
-        key,
-        path: opts.dir,
-        paused: paused[key],
-        ...opts
-      })
+      let datdata
```
Comment (martinheidegger) [inline app/actions/dat-middleware.js]: Is there a reason why you renamed the variable?

Comment (AtuyL) [reply app/actions/dat-middleware.js]: that variable is "not opts". The `opts` have inside of.
below code from `storeOnDisk` method.

```javascript
[key]: {
    dir: this.dats[key].path,
    opts: this.dats[key].opts
}
```

**Pull request closed without merging**